lib/AST/AutoDiff.cpp

```diff
@@ -366,6 +366,13 @@ GenericSignature autodiff::getDifferentiabilityWitnessGenericSignature(
   return derivativeGenSig;
 }
 
+TangentSpace TangentSpace::getTuple(TupleType *tupleTy) {
+  if (tupleTy->getElements().size() == 0) {
+    llvm::report_fatal_error("Attempted to get tangent space of empty tuple.");
+  }
```

Comment:
Note: We mainly use `assert` to stand guard of internal invariants. `llvm::report_fatal_error` is meant for something the *user* is to blame for, like passing a flag and violating its contract.